rust/parquet/src/encodings/decoding.rs

```diff
@@ -49,6 +49,52 @@ pub trait Decoder<T: DataType> {
     /// `buffer.len()`.
     fn get(&mut self, buffer: &mut [T::T]) -> Result<usize>;
 
+    /// Consume values from this decoder and write the results to `buffer`, leaving
+    /// "spaces" for null values.
+    ///
+    /// `null_count` is the number of nulls we expect to see in `buffer`, after reading.
+    /// `valid_bits` stores the valid bit for each value in the buffer. It should contain
+    ///   at least number of bits that equal to `buffer.len()`.
+    ///
+    /// Returns the actual number of values decoded.
+    ///
+    /// # Panics
+    ///
+    /// Panics if `null_count` is greater than `buffer.len()`.
+    fn get_spaced(
+        &mut self,
+        buffer: &mut [T::T],
+        null_count: usize,
+        valid_bits: &[u8],
+    ) -> Result<usize> {
+        assert!(buffer.len() >= null_count);
+
+        // TODO: check validity of the input arguments?
+        if null_count == 0 {
+            return self.get(buffer);
+        }
+
+        let num_values = buffer.len();
+        let values_to_read = num_values - null_count;
+        let values_read = self.get(buffer)?;
+        if values_read != values_to_read {
```

Comment:
What if buffer size is longer than available values?

Reply:
Thanks @liurenjie1024 , and sorry for the late reply. It won't work in that case, since later when setting up the null values the indices will all be wrong. The requirement is that the caller of this method should pass in the correct arguments, based on reading the definition & repetition levels. We need to know the total number of values and the indices of the null values among them. Here, we can optionally pass in the total number of values as a parameter, but that can just be done by using `buffer.len()`.